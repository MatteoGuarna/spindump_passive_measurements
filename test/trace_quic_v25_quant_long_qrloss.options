--report-qr-loss --textual --format json --report-spins --report-notes
