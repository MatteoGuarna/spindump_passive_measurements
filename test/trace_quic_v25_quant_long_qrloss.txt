Draft-25 test with QUIC and Quant, expecting to see some spin bits. Using an additional option for QR/QL loss measurements, making sure that does not disrupt regular QUIC flows without those bits.
