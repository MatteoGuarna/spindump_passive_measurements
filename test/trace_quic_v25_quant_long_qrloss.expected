[
{ "Event": "new", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "04d1fd17-6dbc29852a5c199502fa75bbb1524a542d5a3cdb (62179:4433)", "Ts": 1580737152390036, "State": "Starting", "Notes": "No rsp,v.qn25,0-RTT,no spin", "Packets1": 0, "Packets2": 0, "Bytes1": 0, "Bytes2": 0 },
{ "Event": "change", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "04d1fd17-04e15005 (62179:4433)", "Ts": 1580737152428008, "State": "Starting", "Notes": "No rsp,v.qn25,0-RTT,no spin", "Packets1": 1, "Packets2": 0, "Bytes1": 1228, "Bytes2": 0, "Bandwidth1": 1228, "Bandwidth2": 0 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "04d1fd17-04e15005 (62179:4433)", "Ts": 1580737152428008, "State": "Up", "Notes": "No rsp,v.qn25,0-RTT,no spin", "Right_rtt": 37972, "Packets1": 1, "Packets2": 0, "Bytes1": 1228, "Bytes2": 0, "Bandwidth1": 1228, "Bandwidth2": 0 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152467799, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 2, "Bytes1": 2872, "Bytes2": 1570, "Bandwidth1": 2872, "Bandwidth2": 1570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152467803, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 3, "Bytes1": 2872, "Bytes2": 3070, "Bandwidth1": 2872, "Bandwidth2": 3070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468209, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 4, "Bytes1": 2872, "Bytes2": 4570, "Bandwidth1": 2872, "Bandwidth2": 4570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468214, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 5, "Bytes1": 2872, "Bytes2": 6070, "Bandwidth1": 2872, "Bandwidth2": 6070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468528, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 6, "Bytes1": 2872, "Bytes2": 7570, "Bandwidth1": 2872, "Bandwidth2": 7570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468532, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 7, "Bytes1": 2872, "Bytes2": 9070, "Bandwidth1": 2872, "Bandwidth2": 9070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468533, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 8, "Bytes1": 2872, "Bytes2": 10570, "Bandwidth1": 2872, "Bandwidth2": 10570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468535, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 3, "Packets2": 9, "Bytes1": 2872, "Bytes2": 12070, "Bandwidth1": 2872, "Bandwidth2": 12070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468703, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 1, "Who": "initiator", "Packets1": 3, "Packets2": 10, "Bytes1": 2872, "Bytes2": 13570, "Bandwidth1": 2872, "Bandwidth2": 13570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152468901, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 0, "Who": "responder", "Packets1": 4, "Packets2": 10, "Bytes1": 2956, "Bytes2": 13570, "Bandwidth1": 2956, "Bandwidth2": 13570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152469558, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 1, "Who": "initiator", "Packets1": 4, "Packets2": 11, "Bytes1": 2956, "Bytes2": 15070, "Bandwidth1": 2956, "Bandwidth2": 15070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152495145, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 1, "Who": "initiator", "Packets1": 5, "Packets2": 11, "Bytes1": 3014, "Bytes2": 15070, "Bandwidth1": 3014, "Bandwidth2": 15070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152503376, "State": "Up", "Notes": "V.qn25,0-RTT,no spin", "Value": 1, "Who": "responder", "Packets1": 6, "Packets2": 11, "Bytes1": 3073, "Bytes2": 15070, "Bandwidth1": 3073, "Bandwidth2": 15070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152503685, "State": "Up", "Notes": "V.qn25,0-RTT,no I-spin", "Value": 1, "Who": "responder", "Packets1": 6, "Packets2": 12, "Bytes1": 3073, "Bytes2": 16570, "Bandwidth1": 3073, "Bandwidth2": 16570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152503693, "State": "Up", "Notes": "V.qn25,0-RTT,no I-spin", "Value": 1, "Who": "responder", "Packets1": 6, "Packets2": 13, "Bytes1": 3073, "Bytes2": 18070, "Bandwidth1": 3073, "Bandwidth2": 18070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152504004, "State": "Up", "Notes": "V.qn25,0-RTT,no I-spin", "Value": 1, "Who": "responder", "Packets1": 6, "Packets2": 14, "Bytes1": 3073, "Bytes2": 19570, "Bandwidth1": 3073, "Bandwidth2": 19570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152504613, "State": "Up", "Notes": "V.qn25,0-RTT,no I-spin", "Value": 0, "Who": "initiator", "Packets1": 6, "Packets2": 15, "Bytes1": 3073, "Bytes2": 21070, "Bandwidth1": 3073, "Bandwidth2": 21070 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152504613, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Left_rtt": 1237, "Packets1": 6, "Packets2": 15, "Bytes1": 3073, "Bytes2": 21070, "Bandwidth1": 3073, "Bandwidth2": 21070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152508886, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 15, "Bytes1": 3137, "Bytes2": 21070, "Bandwidth1": 3137, "Bandwidth2": 21070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509263, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 16, "Bytes1": 3137, "Bytes2": 22570, "Bandwidth1": 3137, "Bandwidth2": 22570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509270, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 17, "Bytes1": 3137, "Bytes2": 24070, "Bandwidth1": 3137, "Bandwidth2": 24070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509272, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 18, "Bytes1": 3137, "Bytes2": 25570, "Bandwidth1": 3137, "Bandwidth2": 25570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509274, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 19, "Bytes1": 3137, "Bytes2": 27070, "Bandwidth1": 3137, "Bandwidth2": 27070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509275, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 20, "Bytes1": 3137, "Bytes2": 28570, "Bandwidth1": 3137, "Bandwidth2": 28570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509277, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 21, "Bytes1": 3137, "Bytes2": 30070, "Bandwidth1": 3137, "Bandwidth2": 30070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509278, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 22, "Bytes1": 3137, "Bytes2": 31570, "Bandwidth1": 3137, "Bandwidth2": 31570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509280, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 23, "Bytes1": 3137, "Bytes2": 33070, "Bandwidth1": 3137, "Bandwidth2": 33070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509281, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 7, "Packets2": 24, "Bytes1": 3137, "Bytes2": 34570, "Bandwidth1": 3137, "Bandwidth2": 34570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509676, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 0, "Who": "initiator", "Packets1": 7, "Packets2": 25, "Bytes1": 3137, "Bytes2": 36070, "Bandwidth1": 3137, "Bandwidth2": 36070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152509792, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 8, "Packets2": 25, "Bytes1": 3195, "Bytes2": 36070, "Bandwidth1": 3195, "Bandwidth2": 36070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152510137, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 8, "Packets2": 26, "Bytes1": 3195, "Bytes2": 37570, "Bandwidth1": 3195, "Bandwidth2": 37570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152511242, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 0, "Who": "initiator", "Packets1": 8, "Packets2": 27, "Bytes1": 3195, "Bytes2": 39070, "Bandwidth1": 3195, "Bandwidth2": 39070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152528773, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 9, "Packets2": 27, "Bytes1": 3253, "Bytes2": 39070, "Bandwidth1": 3253, "Bandwidth2": 39070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152528777, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 9, "Packets2": 28, "Bytes1": 3253, "Bytes2": 40570, "Bandwidth1": 3253, "Bandwidth2": 40570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152529206, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 0, "Who": "initiator", "Packets1": 9, "Packets2": 29, "Bytes1": 3253, "Bytes2": 42070, "Bandwidth1": 3253, "Bandwidth2": 42070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152538322, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 0, "Who": "responder", "Packets1": 10, "Packets2": 29, "Bytes1": 3311, "Bytes2": 42070, "Bandwidth1": 3311, "Bandwidth2": 42070 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152538322, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Right_rtt": 33709, "Packets1": 10, "Packets2": 29, "Bytes1": 3311, "Bytes2": 42070, "Bandwidth1": 3311, "Bandwidth2": 42070 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152538322, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Full_rtt_responder": 34946, "Packets1": 10, "Packets2": 29, "Bytes1": 3311, "Bytes2": 42070, "Bandwidth1": 3311, "Bandwidth2": 42070 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152538966, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 0, "Who": "responder", "Packets1": 10, "Packets2": 30, "Bytes1": 3311, "Bytes2": 43570, "Bandwidth1": 3311, "Bandwidth2": 43570 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152539305, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "initiator", "Packets1": 10, "Packets2": 31, "Bytes1": 3311, "Bytes2": 44300, "Bandwidth1": 3311, "Bandwidth2": 44300 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152539305, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Left_rtt": 983, "Packets1": 10, "Packets2": 31, "Bytes1": 3311, "Bytes2": 44300, "Bandwidth1": 3311, "Bandwidth2": 44300 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152539305, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Full_rtt_initiator": 34692, "Packets1": 10, "Packets2": 31, "Bytes1": 3311, "Bytes2": 44300, "Bandwidth1": 3311, "Bandwidth2": 44300 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152540126, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "initiator", "Packets1": 11, "Packets2": 31, "Bytes1": 3369, "Bytes2": 44300, "Bandwidth1": 3369, "Bandwidth2": 44300 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152574060, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Value": 1, "Who": "responder", "Packets1": 12, "Packets2": 31, "Bytes1": 3422, "Bytes2": 44300, "Bandwidth1": 3422, "Bandwidth2": 44300 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152574060, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Right_rtt": 34755, "Packets1": 12, "Packets2": 31, "Bytes1": 3422, "Bytes2": 44300, "Bandwidth1": 3422, "Bandwidth2": 44300 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["100.72.46.172","13.48.145.10"], "Session": "11547fcd-04e15005 (62179:4433)", "Ts": 1580737152574060, "State": "Up", "Notes": "V.qn25,0-RTT,spinning", "Full_rtt_responder": 35738, "Packets1": 12, "Packets2": 31, "Bytes1": 3422, "Bytes2": 44300, "Bandwidth1": 3422, "Bandwidth2": 44300 }
]
